src/agenteval/targets/bedrock_agent/target.py

```diff
@@ -25,35 +32,62 @@ def __init__(self, bedrock_agent_id: str, bedrock_agent_alias_id: str, **kwargs)
         self._session_id: str = str(uuid.uuid4())
 
     def invoke(self, prompt: str) -> TargetResponse:
-        """Invoke the target with a prompt.
+        completion = ""
+        trace_data = []
 
-        Args:
-            prompt (str): The prompt as a string.
+        back_off = 0.01  # In seconds.
+        done = False
+        while not done:
+            try:
```

Comment:
I think the `try`/`except` should be applied only when iterating through the completion `EventStream`. Throttling exceptions from `invoke_agent` should already be handled using [adaptive retry](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/retries.html).